src/test/java/org/folio/rest/api/ScheduledNoticesAPITest.java

```diff
@@ -0,0 +1,308 @@
+import org.folio.rest.support.Response;
+import org.folio.rest.support.ResponseHandler;
+
+public class ScheduledNoticesAPITest extends ApiTests {
+
+  private static final RecurringPeriod ONE_DAY_PERIOD = new RecurringPeriod()
+    .withDuration(1)
+    .withIntervalId(RecurringPeriod.IntervalId.DAYS);
+
+  private static final RecurringPeriod ONE_MONTH_PERIOD = new RecurringPeriod()
+    .withDuration(1)
+    .withIntervalId(RecurringPeriod.IntervalId.MONTHS);
+
+  @Before
+  public void beforeEach() throws MalformedURLException {
+    StorageTestSuite.deleteAll(scheduledNoticesStorageUrl("/scheduled-notices"));
+  }
+
+  @Test
+  public void canSaveScheduledNotice() throws MalformedURLException,InterruptedException, ExecutionException,
+    TimeoutException {
+
+    Date nextRunTime = new Date();
+    NoticeConfig.Timing timing = NoticeConfig.Timing.BEFORE;
+    String templateId = "a0d83326-d47e-43c6-8da6-f972015c3b52";
+    NoticeConfig.Format format = NoticeConfig.Format.EMAIL;
+
+    String createdNoticeId = createScheduledNotice(nextRunTime, timing, ONE_DAY_PERIOD, templateId, format).getId();
+
+    ScheduledNotice createdNotice = getById(scheduledNoticesStorageUrl("/scheduled-notices/" + createdNoticeId))
+      .mapTo(ScheduledNotice.class);
+    NoticeConfig createdConfig = createdNotice.getNoticeConfig();
+
+    assertThat(createdNotice.getNextRunTime(), is(nextRunTime));
+    assertThat(createdConfig.getTiming(), is(timing));
+    assertThat(createdConfig.getRecurringPeriod().getIntervalId(), is(ONE_DAY_PERIOD.getIntervalId()));
+    assertThat(createdConfig.getRecurringPeriod().getDuration(), is(ONE_DAY_PERIOD.getDuration()));
+    assertThat(createdConfig.getTemplateId(), is(templateId));
+    assertThat(createdConfig.getFormat(), is(format));
+  }
+
+  @Test
+  public void canGetScheduledNoticesCollection() throws MalformedURLException, InterruptedException,
+    ExecutionException, TimeoutException {
+
+    createScheduledNotice(NoticeConfig.Timing.BEFORE, ONE_DAY_PERIOD,
+      "a0d83326-d47e-43c6-8da6-f972015c3b52", NoticeConfig.Format.EMAIL);
+
+    createScheduledNotice(NoticeConfig.Timing.AFTER, ONE_MONTH_PERIOD,
+      "b84b721e-76ca-49ef-b486-f3debd92371d", NoticeConfig.Format.SMS);
+
+
+    CompletableFuture<JsonResponse> getCompleted = new CompletableFuture<>();
+    client.get(scheduledNoticesStorageUrl("/scheduled-notices"), TENANT_ID, ResponseHandler.json(getCompleted));
+    JsonResponse response = getCompleted.get(5, SECONDS);
+    ScheduledNotices scheduledNotices = response.getJson().mapTo(ScheduledNotices.class);
+
+    assertThat(scheduledNotices.getScheduledNotices().size(), is(2));
+    assertThat(scheduledNotices.getTotalRecords(), is(2));
+  }
+
+  @Test
+  public void canGetScheduledNoticesCollectionByQuery() throws MalformedURLException, InterruptedException,
+    ExecutionException, TimeoutException {
+
+    String templateId = "b84b721e-76ca-49ef-b486-f3debd92371d";
```

Comment:
I prefer the use of dynamic UUIDs, however I'm aware many folks prefer the hard coded approach

Reply:
I see that in most of the test in mod-circulation-storage random values are used. So I made a changes to use random values in order to keep consistency with other tests.

Reply:
Thank you